### ISSUE TITLE
[Bug]: Windows 11 Client ver 3.10.0 Error file sharing with password

### ISSUE THREAD

**Issue by Niskeletor** [nextcloud/desktop#6081]
### ⚠️ Before submitting, please verify the following: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

Error when trying to share a resource (file or directory) from the Windows 11 file system, by right clicking Nextcloud / Share options.
Password dialog appears, but even if you enter a complex and complete password, it asks for a new password again.

We have the option from the server to Force password request. If I disable this function the resource can be shared [without password]. 

However from the web client this error does not occur. 

![error2](https://github.com/nextcloud/desktop/assets/97298802/56aa78c7-629d-4102-988f-172017ae4f89)
![error01](https://github.com/nextcloud/desktop/assets/97298802/3d3b4405-eee4-41c2-8b80-1d379d35f15f)

### Steps to reproduce

1. Right click on the file
2. Click on Share Options
3. The share dialog does not accept password
...

### Expected behavior

After entering a password that meets the requirements, you should validate or accept it as in the web client.

### Which files are affected by this bug

Any file or directory

### Operating system

Windows

### Which version of the operating system you are running.

Windows 11

### Package

Appimage

### Nextcloud Server version

27.1.0

### Nextcloud Desktop Client version

3.10.0

### Is this bug present after an update or on a fresh install?

Updated from a minor version (ex. 3.4.2 to 3.4.4)

### Are you using the Nextcloud Server Encryption module?

Encryption is Enabled

### Are you using an external user-backend?

- [X] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Nextcloud Server logs

```shell
[no app in context] Error: InvalidArgumentException: Passwords are enforced for link and mail shares at <<closure>>

0. /home/nextcloud/public_html/lib/private/Share20/Manager.php line 779
   OC\Share20\Manager->verifyPassword()
1. /home/nextcloud/public_html/apps/files_sharing/lib/Controller/ShareAPIController.php line 834
   OC\Share20\Manager->createShare("*** sensitive parameters replaced ***")
2. /home/nextcloud/public_html/lib/private/AppFramework/Http/Dispatcher.php line 230
   OCA\Files_Sharing\Controller\ShareAPIController->createShare("*** sensitive parameters replaced ***")
3. /home/nextcloud/public_html/lib/private/AppFramework/Http/Dispatcher.php line 137
   OC\AppFramework\Http\Dispatcher->executeController()
4. /home/nextcloud/public_html/lib/private/AppFramework/App.php line 183
   OC\AppFramework\Http\Dispatcher->dispatch()
5. /home/nextcloud/public_html/lib/private/Route/Router.php line 315
   OC\AppFramework\App::main()
6. /home/nextcloud/public_html/ocs/v1.php line 64
   OC\Route\Router->match()
7. /home/nextcloud/public_html/ocs/v2.php line 23
   require_once("/home/nextcloud/public_html/ocs/v1.php")

POST /ocs/v2.php/apps/files_sharing/api/v1/shares?format=json
from 192.168.177.1 by pablonr at 2023-09-21T12:15:02+02:00
```

**Comment by Spheron101** [comment]
Can confirm this on Windows 10. Nextcloud Desktop Client Updatet from 3.6.6 to 3.10.0. We force the password request at server settings, too.

**Comment by ksainc** [comment]
Confirmed.

Happening on Windows 10 Pro and 11 Pro.

Downgrade to 3.9.4 fixes the issue.

Server Logs,

[no app in context] Error: InvalidArgumentException: Passwords are enforced for link and mail shares at <<closure>>

0. /var/www/nextcloud/lib/private/Share20/Manager.php line 780
   OC\Share20\Manager->verifyPassword()
1. /var/www/nextcloud/apps/files_sharing/lib/Controller/ShareAPIController.php line 834
   OC\Share20\Manager->createShare("*** sensitive parameters replaced ***")
2. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 230
   OCA\Files_Sharing\Controller\ShareAPIController->createShare("*** sensitive parameters replaced ***")
3. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 137
   OC\AppFramework\Http\Dispatcher->executeController()
4. /var/www/nextcloud/lib/private/AppFramework/App.php line 183
   OC\AppFramework\Http\Dispatcher->dispatch()
5. /var/www/nextcloud/lib/private/Route/Router.php line 315
   OC\AppFramework\App::main()
6. /var/www/nextcloud/ocs/v1.php line 64
   OC\Route\Router->match()
7. /var/www/nextcloud/ocs/v2.php line 23
   require_once("/var/www/nextcloud/ocs/v1.php")

POST /ocs/v2.php/apps/files_sharing/api/v1/shares?format=json
from <IP> by <username> at 2023-09-22T22:24:12+00:00

**Comment by Flo-MF** [comment]
Can confirm, that this issue is happening on macOS too.

Client 3.10.0
Nextcloud server-version: 26.0.4.2

**Comment by claucambra** [comment]
Hey all, we have a pending fix for this in #6100
